front/internals/scripts/extract-intl.js

```diff
@@ -58,34 +57,33 @@ const writeFilePromise = (fileName, data) => new Promise((resolve, reject) => {
   });
 });
 
-// Store existing translations into memory
-const oldLocaleMappings = [];
-const localeMappings = [];
-
-// Loop to run once per locale
-for (const locale of locales) {
-  oldLocaleMappings[locale] = {};
-  localeMappings[locale] = {};
-  // File to store translation messages into
-  const translationFileName = `app/translations/${locale}.json`;
-  try {
-    // Parse the old translation message JSON files
-    const messages = JSON.parse(fs.readFileSync(translationFileName, 'utf8'));
-    const messageKeys = Object.keys(messages);
-    for (const messageKey of messageKeys) {
-      oldLocaleMappings[locale][messageKey] = messages[messageKey];
-    }
-  } catch (error) {
-    if (error.code !== 'ENOENT') {
-      process.stderr.write(
-        `There was an error loading this translation file: ${translationFileName}
+const generateLocaleMappings = (outputFolder, oldLocaleMappings, localeMappings) => {
+  // Loop to run once per locale
+  for (const locale of locales) {
+    oldLocaleMappings[locale] = {};
+    localeMappings[locale] = {};
+    // File to store translation messages into
+    const translationFileName = `${outputFolder}${locale}.json`;
+    try {
+      // Parse the old translation message JSON files
+      const messages = JSON.parse(fs.readFileSync(translationFileName, 'utf8'));
+      const messageKeys = Object.keys(messages);
+      for (const messageKey of messageKeys) {
+        oldLocaleMappings[locale][messageKey] = messages[messageKey];
+      }
+    } catch (error) {
+      if (error.code !== 'ENOENT') {
+        process.stderr.write(
+          `There was an error loading this translation file: ${translationFileName}
         \n${error}`
-      );
+        );
+      }
```

Comment:
Stop depending on global variables `oldLocaleMappings` and `localeMappings`, as those are different for non-admin and admin translations. Instead, generate them through a function.